test/e2e/aks.go

```diff
@@ -188,7 +188,7 @@ func WaitForControlPlaneMachinesToExist(ctx context.Context, input WaitForContro
 		ammpList := &infrav1exp.AzureManagedMachinePoolList{}
 
 		if err := input.Lister.List(ctx, ammpList, opt1, opt2); err != nil {
-			Logf("Failed to get machinePool: %+v", err)
+			LogWarningf("Failed to get machinePool: %+v", err)
```

Comment:
side node: why is this func called `WaitForControlPlaneMachinesToExist` seems like it's not checking control planes (that wouldn't even make sense since it's for AKS)

Reply:
I know... (I mean, I don't actually know)

Reply:
seems like either faulty copy/pasta or a "trick" to make it fit into the test framework but then it should really have better comments. My bet is on #1. 

anyways that's out of scope here.